packages/ref-doc/src/extractor.ts

```diff
@@ -140,25 +150,41 @@ function extractOperationRefDoc(program: Program, operation: Operation): Operati
   };
 }
 
-function extractDecoratorRefDoc(program: Program, decorator: Decorator): DecoratorRefDoc {
+function extractDecoratorRefDoc(
+  program: Program,
+  decorator: Decorator
+): [DecoratorRefDoc, TypeSpecRefDocError[]] {
+  const errors: TypeSpecRefDocError[] = [];
   const paramDoc = getParmeterDocs(decorator);
   const parameters: FunctionParameterRefDoc[] = decorator.parameters.map((x) => {
+    const doc = paramDoc.get(x.name);
+    if (doc === undefined) {
+      errors.push({
+        message: `Missing decorator param doc: '${decorator.name}.${x.name}'`,
+      });
+    }
     return {
       type: x,
-      doc: paramDoc.get(x.name) ?? "",
+      doc: doc ?? "",
       name: x.name,
       optional: x.optional,
       rest: x.rest,
     };
   });
 
   const examples = extractExamples(decorator);
-  return {
+  const mainDoc = extractMainDoc(program, decorator);
+  if (mainDoc === undefined || mainDoc === "") {
+    errors.push({
```

Comment:
So the goal at some point is to expose ref-doc as a package anyone can consume. I feel like having an error preventing generation is a bit too restrictive. Maybe we should have `error` and `warning` severity. 

Or we should have this logic in the library-linter

Reply:
I'll try doing it with the library linter. I think that will be easier to implement anyhow.

Reply:
This is the approach I followed (I think) giving ref-doc its own diagnostics and simply emitting those. Let me know what you think.